sys/kern/main.c

```diff
@@ -17,7 +19,20 @@ int kmain(void) {
   proc_add(proc_create(thread_self(), NULL));
 
   if (init) {
-    run_program(init, (char *[]){init, NULL}, (char *[]){NULL});
+    int ninit_args;
+    char **init_args = kenv_get_init_args(&ninit_args);
+
+    if (init_args) {
+      char **args = kmalloc(M_TEMP, sizeof(char *) * (ninit_args + 2), 0);
```

Comment:
Shouldn't this memory leak be handled?

Reply:
Done. (d6fd9d470780afc6d27bc4f28035418f04d7a828)